seed/seedwriter/helpers.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package seedwriter
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/snap"
+)
+
+// A TrailFetcher is a Fetcher that can return the trail of references
+// of fetched assertions.
+type TrailFetcher interface {
```

Comment:
(bikesheed) Maybe a name that relates a bit more to assertions? TrailFetcher is very generic. Maybe `AssertsRefFetcher, RefFetcher ` and if we go with that we could rename "Trail" to something like "Refs()".

Reply:
Fair, we have only one Fetcher interface atm in asserts, but I agree. Will see what to do.